src/Core/Content/Rule/DataAbstractionLayer/RulePayloadUpdater.php

```diff
@@ -102,6 +122,135 @@ public function updatePayloads(EntityWrittenEvent $event): void
         $this->update(Uuid::fromBytesToHexList($ruleIds));
     }
 
+    /**
+     * @throws UnsupportedCommandTypeException
+     */
+    public function triggerChangeset(PreWriteValidationEvent $event): void
+    {
+        $commands = $event->getCommands();
+
+        foreach ($commands as $command) {
+            if ($command->getDefinition()->getClass() !== RuleConditionDefinition::class) {
+                continue;
+            }
+
+            if ($command instanceof DeleteCommand) {
+                $command->requestChangeSet();
+
+                continue;
+            }
+
+            if ($command instanceof InsertCommand) {
+                continue;
+            }
+
+            if ($command instanceof UpdateCommand) {
+                if ($command->hasField('value')) {
+                    $value = json_decode($command->getPayload()['value'], true);
```

Comment:
throw on error

Reply:
I opted to skip requesting the changeset, if it's not the value we need, instead.